runtime/intern.c

```diff
@@ -675,11 +677,32 @@ static void intern_add_to_heap(mlsize_t whsize)
     }
     caml_allocated_words +=
       Wsize_bsize ((char *) intern_dest - intern_extra_block);
-    caml_add_to_heap(intern_extra_block);
+    if(caml_add_to_heap(intern_extra_block) != 0) {
```

Comment:
This OOM check is new. Was this just wrong before?

Reply:
Indeed.

Reply:
Do you think this deserves its own PR? This bug is quite difficult to trigger...

Reply:
No, I think it's fine as part of this one.